source/extensions/filters/network/redis_proxy/command_splitter_impl.cc

```diff
@@ -463,18 +513,55 @@ SplitRequestPtr InstanceImpl::makeRequest(Common::Redis::RespValuePtr&& request,
     return nullptr;
   }
 
+  // Get the handler for the downstream request
   auto handler = handler_lookup_table_.find(to_lower_string.c_str());
   if (handler == nullptr) {
     stats_.unsupported_command_.inc();
     callbacks.onResponse(Common::Redis::Utility::makeError(
         fmt::format("unsupported command '{}'", request->asArray()[0].asString())));
     return nullptr;
   }
+
+  // Fault Injection Check
+  Common::Redis::FaultSharedPtr fault_ptr = fault_manager_->getFaultForCommand(to_lower_string);
```

Comment:
Does this need to return a shared_ptr? Can you just return a raw pointer for perf reasons?

Reply:
I'll do that.